app/src/main/java/de/westnordost/streetcomplete/overlays/cycleway/StreetCyclewayOverlayForm.kt

```diff
@@ -95,56 +113,71 @@ class StreetCyclewayOverlayForm : AStreetSideSelectOverlayForm<CyclewayAndDirect
         streetSideSelect.setPuzzleSide(rightItem, true)
     }
 
-    /* ----------------------------------- bicycle boulevards ----------------------------------- */
+    /* ------------------------- pedestrian zone and bicycle boulevards ------------------------- */
 
-    private fun createSwitchBicycleBoulevardAnswer(): IAnswerItem? =
-        if (bicycleBoulevard == BicycleBoulevard.YES) {
-            AnswerItem2(
-                getString(R.string.bicycle_boulevard_is_not_a, getString(R.string.bicycle_boulevard)),
-                ::removeBicycleBoulevard
-            )
-        } else if (countryInfo.hasBicycleBoulevard) {
-            AnswerItem2(
-                getString(R.string.bicycle_boulevard_is_a, getString(R.string.bicycle_boulevard)),
-                ::addBicycleBoulevard
-            )
-        } else {
-            null
-        }
+    private fun createSwitchBicycleInPedestrianZoneAnswers(): List<IAnswerItem> {
+        if (bicycleInPedestrianStreet == null) return listOf()
 
-    private fun removeBicycleBoulevard() {
-        bicycleBoulevard = BicycleBoulevard.NO
-        updateBicycleBoulevard()
+        val result = mutableListOf<IAnswerItem>()
+        if (bicycleInPedestrianStreet != BicycleInPedestrianStreet.DESIGNATED) {
+            result.add(AnswerItem(R.string.pedestrian_zone_designated) {
+                bicycleInPedestrianStreet = BicycleInPedestrianStreet.DESIGNATED
+                updateStreetSign()
+            })
+        }
+        if (bicycleInPedestrianStreet != BicycleInPedestrianStreet.ALLOWED) {
+            result.add(AnswerItem(R.string.pedestrian_zone_allowed_sign) {
+                bicycleInPedestrianStreet = BicycleInPedestrianStreet.ALLOWED
+                updateStreetSign()
+            })
+        }
+        if (bicycleInPedestrianStreet != BicycleInPedestrianStreet.NOT_SIGNED) {
+            result.add(AnswerItem(R.string.pedestrian_zone_no_sign) {
+                bicycleInPedestrianStreet = BicycleInPedestrianStreet.NOT_SIGNED
+                updateStreetSign()
+            })
+        }
+        return result
```

Comment:
Would you consider this to be more readable?

```kt
        return listOfNotNull(
            AnswerItem(R.string.pedestrian_zone_designated) {
                bicycleInPedestrianStreet = BicycleInPedestrianStreet.DESIGNATED
                updateStreetSign()
            }.takeIf { bicycleInPedestrianStreet != BicycleInPedestrianStreet.DESIGNATED },

            AnswerItem(R.string.pedestrian_zone_allowed_sign) {
                bicycleInPedestrianStreet = BicycleInPedestrianStreet.ALLOWED
                updateStreetSign()
            }.takeIf { bicycleInPedestrianStreet != BicycleInPedestrianStreet.ALLOWED },

            AnswerItem(R.string.pedestrian_zone_no_sign) {
                bicycleInPedestrianStreet = BicycleInPedestrianStreet.NOT_SIGNED
                updateStreetSign()
            }.takeIf { bicycleInPedestrianStreet != BicycleInPedestrianStreet.NOT_SIGNED },
        )
```

Reply:
a little bit, I'll take it!